openedx/openedx-platform
# revert: temp: let XBlock API users optionally use LabXchange block types

## Description

Reverts ef8f841ac276eec2609f6960a963a790dcd53f43, which added Labxchange-specific code allowing LabXchange XBlocks to be used when loading and rendering non-lx blocks.

This code is no longer required, as LabXchange are storing OLX in their own database, and not using Open edX for rendering anymore. Some Content Libraries (Blockstore) calls are still being made to fetch asset files, but these do not require loading blocks, and so the LX-specific code here (and dependencies in https://github.com/edx/edx-internal/pull/7420) can be removed from edx.org without adversely impacting LabXchange.org.

This change impacts LabXchange only, and should not have any effect on other users of Open edX.

## Supporting information

See https://github.com/openedx/edx-platform/pull/29517

## Testing instructions

CI should be sufficient to ensure this change is safe for Open edX.
LabXchange testing will take place on (internal) https://gitlab.com/opencraft/client/LabXchange/labxchange-dev/-/merge_requests/1862.

## Deadline

None
